File: src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Weather.java
package io.github.amerebagatelle.fabricskyboxes.util.object;

import com.google.common.collect.ImmutableMap;
import com.mojang.serialization.Codec;

import java.util.Map;
import java.util.Objects;

public enum Weather {
    CLEAR("clear"),
    RAIN("rain"),
    BIOME_RAIN("rain_biome"),
    SNOW("snow"),
    THUNDER("thunder");

    private static final Map<String, Weather> VALUES;
    public static final Codec<Weather> CODEC = Codec.STRING.xmap(Weather::fromString, Weather::toString);

    static {
        ImmutableMap.Builder<String, Weather> builder = ImmutableMap.builder();
        for (Weather value : values()) {
            builder.put(value.name, value);
        }
        VALUES = builder.build();
    }

    private final String name;

    Weather(String name) {
        this.name = name;
    }

    public static Weather fromString(String name) {
        return Objects.requireNonNull(VALUES.get(name));
    }

    @Override
    public String toString() {
        return this.name;
    }
}

File: src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/Weather.kt
package io.github.amerebagatelle.fabricskyboxes.util.`object`

enum class Weather(
    val weather: String?
) {
    CLEAR("clear"),
    RAIN("rain"),
    SNOW("snow"),
    THUNDER("thunder");

    companion object {
        fun from(value: String): Weather? = Weather.values().firstOrNull { it.weather == value}
    }
}
